File: AndroidStudio/Login/LoginActivity.java
package com.example.stockapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.HttpHeaderParser;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONException;
import org.json.JSONObject;

import java.net.CookieHandler;
import java.net.CookieManager;
import java.util.List;
import java.util.Map;

public class LoginActivity extends AppCompatActivity {
    public static Context context;

    private EditText et_id, et_pass;
    private Button btn_login, btn_register;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        context = getApplicationContext();

        et_id = findViewById(R.id.et_id);
        et_pass = findViewById(R.id.et_pass);
        btn_login = findViewById(R.id.btn_login);
        btn_register = findViewById(R.id.btn_register);

        // 회원가입 버튼을 클릭 시 수행
        btn_register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(LoginActivity.this, RegisterActivity.class);
                startActivity(intent);
            }
        });

        btn_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                //EditText에 현재 입력되어 있는 값을 get해온다.
                String userID = et_id.getText().toString();
                String userPass = et_pass.getText().toString();

                Response.Listener<String> responseListener = new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        try {
                            Log.d("hahaha", response);
                            JSONObject jsonObject = new JSONObject(response);
                            String responseWord = jsonObject.getString("response");
                            Log.d("hahaha", responseWord);
                            if (responseWord.equals("success_login")) { // 로그인에 성공한 경우

                                Log.d("hahaha", "로그인 성공!!!!!");



                                Toast.makeText(getApplicationContext(), "로그인에 성공하였습니다.", Toast.LENGTH_SHORT).show();

                                SharedPreferences pref = context.getSharedPreferences("sessionCookie",Context.MODE_PRIVATE);
                                String sessionid = pref.getString("sessionid",null);
                                Log.d("hahaha", "여기요!!!!!" + sessionid);

                                Intent intent = new Intent(LoginActivity.this, InfoActivity.class);
                                intent.putExtra("userID", userID);
                                intent.putExtra("sessionid", sessionid);
//                                intent.putExtra("userPass", userPass);
                                startActivity(intent);
                            } else { // 로그인에 실패한 경우
                                Toast.makeText(getApplicationContext(), "로그인에 실패하였습니다.", Toast.LENGTH_SHORT).show();
                                return;
                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }
                };
                LoginRequest loginRequest = new LoginRequest(userID, userPass, responseListener);
                RequestQueue queue = Volley.newRequestQueue(LoginActivity.this);
                queue.add(loginRequest);
            }
        });
    }
}

File: AndroidStudio/retrofit/LoginActivity.kt
package com.chocobi.groot.view.login

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import com.chocobi.groot.MainActivity
import com.chocobi.groot.R
import com.chocobi.groot.view.signup.SignupActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

//        sharedPreference
        val shared = getSharedPreferences("SharedPref", Context.MODE_PRIVATE)
        val editor = shared.edit() // 수정을 위한 에디터

//        초기 토큰값 확인
        var access_token = shared.getString("access_token", "초기값")
        var dialog = AlertDialog.Builder(this@LoginActivity)
        dialog.setTitle("초기값")
        dialog.setMessage(access_token)
        dialog.show()



//        retrofit 객체 만들기
        var retrofit = Retrofit.Builder()
            .baseUrl("http://10.0.2.2:8000")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

//        service 객체 만들기
        var loginService = retrofit.create(LoginService::class.java)


        var loginIdInput = findViewById<EditText>(R.id.loginIdInput)
        var loginPwInput = findViewById<EditText>(R.id.loginPwInput)
        val basicLoginBtn = findViewById<Button>(R.id.basicLoginBtn)
        val toSignupText = findViewById<TextView>(R.id.toSignupText)


//        로그인 버튼 클릭시
        basicLoginBtn.setOnClickListener {
            var textId = loginIdInput.text.toString()
            var textPw = loginPwInput.text.toString()

//            로그인 요청 보내기
            loginService.requestLogin(textId, textPw).enqueue(object:Callback<Login>{

                override fun onResponse(call: Call<Login>, response: Response<Login>) {
//                    통신 성공시 실행되는 코드
                    var login = response.body()

                    var dialog = AlertDialog.Builder(this@LoginActivity)
                    dialog.setTitle("알림!")

//                    토큰 저장
                    editor.putString("access_token", login?.accessToken)
//                    editor.putString("refresh_token", login?.refreshToken)
                    editor.commit()

//                    토큰 확인
                    access_token = shared.getString("access_token", "")
                    dialog.setMessage(access_token)
                    dialog.show()
                }

                override fun onFailure(call: Call<Login>, t: Throwable) {
//                    통신 실패시 실행되는 코드
                    var dialog = AlertDialog.Builder(this@LoginActivity)
                    dialog.setTitle("실패!")
                    dialog.setMessage(t.message)
                    dialog.show()
                }
            })
        }

//        회원가입 안내 텍스트 클릭시
        toSignupText.setOnClickListener {
            var intent = Intent(this, SignupActivity::class.java)
            startActivity(intent)
        }

        //        회원가입 안내 텍스트 클릭시
        toSignupText.setOnClickListener {
            var intent = Intent(this, SignupActivity::class.java)
            startActivity(intent)
        }



    }
}